### ISSUE TITLE
omni shouldn't fail when agg_nick_cache can't be downloaded

### ISSUE THREAD

**Issue by seledwards** [GENI-NSF/geni-tools#662]
`omni` is failing in cases where `agg_nick_cache` can't be downloaded and there is no existing `agg_nick_cache` (usually in `~/.gcf/agg_nick_cache`).

This is a regression which must have appeared when `omni.py` was moved to `oscript.py`.    

The code should always fall back to using `agg_nick_cache.base` distributed with the release, but the path to this file changed with the move to using `oscript.py`.

The fix is to change:
```
    parent_dir = curr_dir.rsplit(os.sep,1)[0]
```
to:
```
    parent_dir = curr_dir.rsplit(os.sep,2)[0]
```

 _Imported from trac ticket [#662](http://trac.gpolab.bbn.com/gcf/ticket/662),  created by sedwards on 07-07-2014 at 11:20, last modified: 07-07-2014 at 12:43_
   CCing: gpo-expt-support@...

**Comment by innnkkki** [comment]
A thing we noticed is that we manually downloaded the agg_nick_cache and put it in ~/.gcf omni was deleting the file and not being able to download it. Maybe if the file can't be downloaded do not delete the existing copy. 

_Trac comment by nriga (github user: innnkkki) on 07-07-2014 at 11:47_

**Comment by seledwards** [comment]
Behavior before the fix is as follows:

With no existing downloaded `agg_nick_cache`:

```
$ ls -l ~/.gcf/agg_nick_cache
ls: /Users/sedwards/.gcf/agg_nick_cache: No such file or directory
```

I get the following (FAILS):

```
 omni.py getversion -a gpo-ig --debug -o --NoAggNickCache --AggNickDefinitiveLocation=foo
11:52:12 INFO    : Attempted to download latest `agg_nick_cache` from 'foo' but could not.
11:52:12 DEBUG   : [Errno 2] No such file or directory: 'foo'
Failed to read any possible agg_nick_cache file.
```

With an `agg_nick_cache` older than a day:

```
$ ls -l ~/.gcf/agg_nick_cache
ls: /Users/sedwards/.gcf/agg_nick_cache: No such file or directory
$ cp agg_nick_cache.base ~/.gcf/agg_nick_cache
$ ls -l ~/.gcf/agg_nick_cache
-rw-r--r--  1 sedwards  sedwards  31939 Jul  7 11:52 /Users/sedwards/.gcf/agg_nick_cache
$ touch -A -990000 ~/.gcf/agg_nick_cache
$ ls -l ~/.gcf/agg_nick_cache
-rw-r--r--  1 sedwards  sedwards  31939 Jul  3 08:52 /Users/sedwards/.gcf/agg_nick_cache
```

I get the following (WORKS):

```
$ omni.py getversion -a gpo-ig --debug -o --NoAggNickCache --AggNickDefinitiveLocation=foo
11:52:44 INFO    : Attempted to download latest `agg_nick_cache` from 'foo' but could not.
11:52:44 DEBUG   : [Errno 2] No such file or directory: 'foo'
11:52:44 INFO    : Loading agg_nick_cache file '/Users/sedwards/.gcf/agg_nick_cache'
<snip>
11:52:45 INFO    :  ------------------------------------------------------
11:52:45 INFO    :  Completed getversion:

  Options as run:
		aggNickDefinitiveLocation: foo
		aggregate: ['gpo-ig']
		debug: True
		framework: portal
		noAggNickCache: True
		noGetVersionCache: True
		output: True
		project: tutorial

  Args: getversion

  Result Summary: Saved getversion at AM gpo-ig to file 'getversion-instageni-gpolab-bbn-com.json'.

Got version for gpo-ig

11:52:45 INFO    :  ======================================================
```

With an `agg_nick_cache` less than a day old:

```
$ cp agg_nick_cache.base ~/.gcf/agg_nick_cache
$ ls -l ~/.gcf/agg_nick_cache
-rw-r--r--  1 sedwards  sedwards  31939 Jul  7 11:54 /Users/sedwards/.gcf/agg_nick_cache```
}}}

I get the following (WORKS):

```
$ omni.py getversion -a gpo-ig --debug -o --NoAggNickCache --AggNickDefinitiveLocation=foo
11:55:43 INFO    : Attempted to download latest `agg_nick_cache` from 'foo' but could not.
11:55:43 DEBUG   : [Errno 2] No such file or directory: 'foo'
11:55:43 INFO    : Loading agg_nick_cache file '/Users/sedwards/.gcf/agg_nick_cache'
<snip>
11:55:44 INFO    :  ------------------------------------------------------
11:55:44 INFO    :  Completed getversion:

  Options as run:
		aggNickDefinitiveLocation: foo
		aggregate: ['gpo-ig']
		debug: True
		framework: portal
		noAggNickCache: True
		noGetVersionCache: True
		output: True
		project: tutorial

  Args: getversion

  Result Summary: Saved getversion at AM gpo-ig to file 'getversion-instageni-gpolab-bbn-com.json'.

Got version for gpo-ig

11:55:44 INFO    :  ======================================================
```

_Trac comment by sedwards (github user: seledwards) on 07-07-2014 at 11:54_

**Comment by seledwards** [comment]
Replying to [nriga](comment:1):
> A thing we noticed is that we manually downloaded the agg_nick_cache and put it in ~/.gcf omni was deleting the file and not being able to download it. Maybe if the file can't be downloaded do not delete the existing copy. 

Hum... That's a windows workaround:
```
            # On Windows, rename doesn't delete any existing file, so explicitly delete the old one first
            # And shutil.move also wants the destination to be gone
            try:
                os.unlink(opts.aggNickCacheName)
            except:
                pass
            shutil.move(tmpcache, opts.aggNickCacheName)
            logger.info("Downloaded latest `agg_nick_cache` from '%s' and copied to '%s'." % (opts.aggNickDefinitiveLocation, opts.aggNickCacheName))
```

_Trac comment by sedwards (github user: seledwards) on 07-07-2014 at 11:58_

**Comment by seledwards** [comment]
The above three tests succeed with the fix in place.

If I cause it to fail before the file is moved (by manually inserting an exception), omni succeeds but still deletes the manually downloaded file.

_Trac comment by sedwards (github user: seledwards) on 07-07-2014 at 12:13_

**Comment by seledwards** [comment]
As the comment in the code says `shutil.move` doesn't work with there is a file in the destination.

[Python docs](https://docs.python.org/2/library/shutil.html#shutil.move) say:

"The destination directory must not already exist."

I think in order to fix the issue with removing the existing file, I would have to add an OS dependency to omni (where none currently exist).  I'm creating a separate ticket to track this issue.

_Trac comment by sedwards (github user: seledwards) on 07-07-2014 at 12:28_

**Comment by seledwards** [comment]
Replying to [sedwards](comment:5):
> As the comment in the code says `shutil.move` doesn't work with there is a file in the destination.
> 
> [Python docs](https://docs.python.org/2/library/shutil.html#shutil.move) say:
> 
> "The destination directory must not already exist."
> 
> I think in order to fix the issue with removing the existing file, I would have to add an OS dependency to omni (where none currently exist).  I'm creating a separate ticket to track this issue.

I made #663 to track making the copy being atomic.  Also more careful reading of #646 indicates the the shutil.move is the fix to #646 and likely #661. 

The fix for this ticket is merged onto develop. Closing.

_Trac comment by sedwards (github user: seledwards) on 07-07-2014 at 12:43_
